### PR TITLE
Add PoL

### PR DESCRIPTION
**Pull request by zeegomo**
Promote PoL and CL crates from nomos-pocs.
CL is mostly unchanged, while PoL has gone through some refactors. I didn't include the crate in the workspace as it has special build instruction and  e.g. does not support clippy

### CODE REVIEW

**Review by danielSanchezQ — APPROVED** [review]
Looking good!
CI is failing in the risc0 tests though

File: cl/cl/src/merkle.rs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        assert!(i < N);
+        leaves[i] = leaf(element);
+    }
+
+    leaves
+}
+
+pub fn leaf(data: &[u8]) -> [u8; 32] {
+    let mut hasher = Sha256::new();
+    hasher.update(b"NOMOS_MERKLE_LEAF");
```
Comment (thomaslavaur) [inline cl/cl/src/merkle.rs:17]: We should consider moving to tag codes if necessary. In Merkle tree constraints, that could save a lot of cycles.

**Review by thomaslavaur — APPROVED** [review]
LGTM

File: cl/cl/src/merkle.rs (hunk 1 above)
Comment (zeegomo) [reply cl/cl/src/merkle.rs:17]: Could you elaborate?

**Review by bacv — APPROVED** [review]

**Pull request merged**